launch/pipeline/service.py

```diff
@@ -1,18 +1,38 @@
 import inspect
+import uuid
 from typing import Any, Callable, Dict, List, Optional
 
 from launch.pipeline.deployment import Deployment
 from launch.pipeline.runtime import Runtime
 
+MAX_SERVICE_NAME_SIZE = 50
+
 
 class ServiceDescription:
     """
     Base ServiceDescription.
     """
 
+    service_name: str
+    service: Callable
+    runtime: Runtime
+    deployment: Deployment
+    env_params: Dict[str, str]
+
+    def __init__(self):
+        self.make_request: Optional[Callable] = None
+
+    def set_make_request(
+        self, make_request: Callable, force: bool = False
+    ) -> None:
+        raise NotImplementedError()
+
     def call(self, req: Any) -> Any:
         raise NotImplementedError()
 
+    def __call__(self, req: Any) -> Any:
```

Comment:
Not sure how I feel about having two ways of calling this thing 🤔

Reply:
we'll this was actually done to avoid any changes on the backend side: otherwise it'd need to use extra logic, i.e. if this is the type, then call "call". Otherwise `__call__`. Probably we should get rid of `.call()`?